gotext.go

```diff
@@ -240,62 +284,79 @@ func GetNDC(dom, str, plural string, n int, ctx string, vars ...interface{}) str
 
 	// Return Translation
 	globalConfig.RLock()
-	tr := globalConfig.storage.GetNDC(dom, str, plural, n, ctx, vars...)
-	globalConfig.RUnlock()
+	defer globalConfig.RUnlock()
 
+	var tr string
+	for i, locale := range globalConfig.locales {
+		if !locale.IsTranslatedNDC(dom, str, n, ctx) && i < (len(globalConfig.locales)-1) {
+			continue
+		}
+		tr = locale.GetNDC(dom, str, plural, n, ctx, vars...)
+		break
+	}
 	return tr
 }
 
-// IsTranslated reports whether a string is translated
-func IsTranslated(str string) bool {
-	return IsTranslatedND(GetDomain(), str, 0)
+// IsTranslated reports whether a string is translated in given language
+func IsTranslated(str, lang string) bool {
```

Comment:
I'd make all new `lang` params optional, so if not passed, will assume the first language in the `languages` list, this way you don't break backwards compatibility.